src/App.jsx

```diff
@@ -14,57 +15,53 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [good, setGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  const handleSelect = item => {
+    setGood(prevGood => (prevGood === item ? '' : item));
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {good ? `${good} is selected` : 'No goods selected'}
+        {good && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => handleSelect('')}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(item => (
+            <tr
+              data-cy="Good"
+              key={item}
+              className={item === good ? 'has-background-success-light' : ''}
+              onClick={() => handleSelect(item)}
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={item === good ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  className={`button ${item === good && 'is-info'}`}
```

Comment:
The `className` assignment uses a template literal with a conditional expression. If `item === good` is false, it will result in `className='button false'`. Consider using a ternary operator to ensure the class is applied correctly: `className={`button ${item === good ? 'is-info' : ''}`}`.